File: chrome/browser/status_icons/status_tray.cc
// Copyright (c) 2012 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/browser/status_icons/status_tray.h"

#include <algorithm>

#include "chrome/browser/status_icons/status_icon.h"

StatusTray::~StatusTray() {
}

StatusIcon* StatusTray::CreateStatusIcon(StatusIconType type,
                                         const gfx::ImageSkia& image,
                                         const string16& tool_tip) {
  StatusIcon* icon = CreatePlatformStatusIcon(type, image, tool_tip);
  if (icon)
    status_icons_.push_back(icon);
  return icon;
}

void StatusTray::RemoveStatusIcon(StatusIcon* icon) {
  StatusIcons::iterator i(
      std::find(status_icons_.begin(), status_icons_.end(), icon));

  if (i == status_icons_.end()) {
    NOTREACHED();
    return;
  }

  status_icons_.erase(i);
}

StatusTray::StatusTray() {
}

File: chrome/android/java/src/org/chromium/chrome/browser/StatusTray.java
// Copyright (c) 2013 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser;

import android.content.Context;

import org.chromium.base.CalledByNative;

/**
 * Java representation of native StatusTray.
 */
public class StatusTray {

    private static StatusTrayFactory sFactory;

    public StatusTray() {
    }

    /**
     * Sets the Factory instance.
     * @param factory The factory instance to set.
     */
    public static void setFactory(StatusTrayFactory factory) {
        sFactory = factory;
    }

    /**
     * Returns a new instance of StatusTray created by a factory instance.
     * @param context Current application context.
     */
    @CalledByNative
    private static StatusTray create(Context context) {
        assert sFactory != null;
        return sFactory.create(context);
    }

    /**
     * Creates the notification to indicate media capture has started.
     */
    @CalledByNative
    protected void createMediaCaptureStatusNotification() {
    }

    /**
     * A factory class to create a StatusTray.
     */
    public static class StatusTrayFactory {
        /**
         * Returns a new instance of StatusTray.
         * @param context Current application context.
         */
        public StatusTray create(Context context) {
            return new StatusTray();
        }
    }

}
